### PR TITLE
CgmesExport. Export transformers with sorted ends

### PR DESCRIPTION
**Pull request by marqueslanauja**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

In Cgmes export, impedances are specified at `powerTransformerEnd1` and powerTransformer preserves the orientation of the transformer in the iidm model.

**What is the new behavior (if this is a feature change)?**
In Cgmes export, high voltage winding must always be `powerTransformerEnd1`. Impedances must always be specified at `powerTransformerEnd1`.
In twoWindingsTransformers oriented, in the iidm model, from low to high nominal voltage, the powerTransformerEnd1 must be mapped to the side 2 of the transformer and must specify the impedances. As the shunt admittances of the transformer are defined, in iidm model, on the side 1, the exported cgmes and iidm models will not be equivalent when the shunt admittances are not zero.
 In threeWindingsTransformers endNumber 1, 2, and 3 are mapped to high, medium and low voltage.

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

**Review by zamarrenolm — APPROVED** [review]

**Comment by colinepiloquet** [comment]
I tested it with the SUV and this fixed most of the warnings. 
Note that in QoCDC v3.3, rules PowerTransformerEndX and PowerTransformerEndR indicated that R and X values on end 1 should be greater than 0.01, which is not always the case in our IGM.
This feature should also probably be optional and by default not activated.

**Comment by marqueslanauja** [comment]
> I tested it with the SUV and this fixed most of the warnings. Note that in QoCDC v3.3, rules PowerTransformerEndX and PowerTransformerEndR indicated that R and X values on end 1 should be greater than 0.01, which is not always the case in our IGM. This feature should also probably be optional and by default not activated.

We can do it in a separated PR if you agree.

**Comment by colinepiloquet** [comment]
> > I tested it with the SUV and this fixed most of the warnings. Note that in QoCDC v3.3, rules PowerTransformerEndX and PowerTransformerEndR indicated that R and X values on end 1 should be greater than 0.01, which is not always the case in our IGM. This feature should also probably be optional and by default not activated.
> 
> We can do it in a separated PR if you agree.

Do you mean adding a parameter to make it optional?

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesExport.java [hunk 1]
```diff
@@ -252,6 +253,12 @@ public String getFormat() {
             "Model description",
             null);
 
+    private static final Parameter EXPORT_TRANSFORMERS_WITH_HIGHEST_VOLTAGE_AT_END1_PARAMETER = new Parameter(
+            EXPORT_TRANSFORMERS_WITH_HIGHEST_VOLTAGE_AT_END1,
+            ParameterType.BOOLEAN,
+            "Export transformers with highest voltage at end1",
+            CgmesExportContext.EXPORT_POWER_FLOWS_FOR_SWITCHES_DEFAULT_VALUE);
+
```
Comment (zamarrenolm) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesExport.java:261]: we should define a separate variable for the default value of this parameter

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/CgmesExportContext.java [hunk 2]
```diff
@@ -57,11 +57,11 @@ public class CgmesExportContext {
 
     public static final boolean EXPORT_BOUNDARY_POWER_FLOWS_DEFAULT_VALUE = true;
     public static final boolean EXPORT_POWER_FLOWS_FOR_SWITCHES_DEFAULT_VALUE = true;
+    public static final boolean EXPORT_TRANSFORMERS_WITH_HIGHEST_VOLTAGE_AT_END1_DEFAULT_VALUE = false;
```
Comment (zamarrenolm) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/CgmesExportContext.java:60]: Ok, the specific variable is defined, but not used in the definition of the `Parameter`.

**Review by zamarrenolm — CHANGES_REQUESTED** [review]
In `CgmesExport` we should set the value for the new parameter when we build the context, like we do with the rest of configurable behaviour (add a line following the pattern for lines 72-77 of `CgmesExport.java`). If we do not set it in the context, the parameter specified through configuration files or properties will not arrive at the equipment export module.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesExport.java (hunk 1 above)
Comment (annetill) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesExport.java:262]: Yes, right.

**Review by zamarrenolm — APPROVED** [review]

**Review by annetill — APPROVED** [review]

**Pull request merged**